stellar-contract-env-common/src/raw_val.rs

```diff
@@ -61,7 +61,18 @@ impl AsMut<RawVal> for RawVal {
 // away, the same way `()` would, while remaining a separate type to allow
 // conversion to a more-structured error code at a higher level.
 #[derive(Debug)]
-pub struct ConversionError;
+pub struct ConversionError<F, T>;
+
+impl<F, T> core::fmt::Display for ConversionError<F, T> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "Conversion from {} to {} failed",
+            std::any::type_name::<F>(),
+            std::any::type_name::<T>()
```

Comment:
In my suggestion below what I mean is make it something like this:
```suggestion
            "Conversion from {} to {} failed",
            self.source,
            std::any::type_name::<T>()
```